src/video.c

```diff
@@ -90,20 +95,29 @@ static VideoCapabilityState video_query_capability_webcanvas(VideoCapability cap
 	}
 }
 
-static void video_add_mode(int width, int height) {
-	if(video.modes) {
-		for(uint i = 0; i < video.mcount; ++i) {
-			VideoMode *m = video.modes + i;
+static void video_add_mode_handler(VideoModeArray *mode_array, int width, int height) {
+	if(mode_array->mode) {
+		for(uint i = 0; i < mode_array->mcount; ++i) {
+			VideoMode *m = mode_array->mode + i;
 
 			if(m->width == width && m->height == height) {
 				return;
 			}
 		}
 	}
 
-	video.modes = (VideoMode*)realloc(video.modes, (++video.mcount) * sizeof(VideoMode));
-	video.modes[video.mcount-1].width  = width;
-	video.modes[video.mcount-1].height = height;
+	mode_array->mode = realloc(mode_array->mode, (++mode_array->mcount) * sizeof(VideoMode));
+
+	mode_array->mode[mode_array->mcount-1].width  = width;
+	mode_array->mode[mode_array->mcount-1].height = height;
+}
+
+static void video_add_mode(int width, int height, bool fullscreen) {
+	if(fullscreen) {
+		video_add_mode_handler(&video.fs_modes, width, height);
+	} else {
+		video_add_mode_handler(&video.win_modes, width, height);
+	}
 }
```

Comment:
I'd rather have separate `video_{add,get}_{fullscreen,windowed}_mode` functions than branch on a boolean like that. It's not clear what `video_add_mode(800, 600, true)` means without context.

Reply:
I was worried about duplicating code, but if separating them makes it easier, I'm totally fine with that.